### PR TITLE
:seedling: Unit tests for binary_artifact

### PR DESCRIPTION
**Pull request by naveensrinivasan**
* **Please check if the PR fulfills these requirements**
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] PR title follows the guidelines defined in  https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**
    https://github.com/ossf/scorecard/issues/986

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Other information**:

### CODE REVIEW

File: checks/raw/binary_artifact_test.go [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+			logger, err := githubrepo.NewLogger(log.DebugLevel)
+			if err != nil {
+				t.Errorf("githubrepo.NewLogger: %v", err)
+			}
+
+			// nolint
+			defer logger.Zap.Sync()
+
+			ctrl := gomock.NewController(t)
+			repo, err := localdir.MakeLocalDirRepo(tt.inputFolder)
```
Comment (azeemshaikh38) [inline checks/raw/binary_artifact_test.go:72]: Why are we creating LocalDir client here? Shouldn't we be using the MockClient?

**Review by azeemshaikh38 — APPROVED** [review]

File: checks/raw/binary_artifact_test.go [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+			}
+
+			ctx := context.Background()
+
+			client := localdir.CreateLocalDirClient(ctx, logger)
+			if err := client.InitRepo(repo); err != nil {
+				t.Errorf("InitRepo: %v", err)
+			}
+			_, err = BinaryArtifacts(client)
+			if !errors.Is(err, tt.err) {
```
Comment (laurentsimon) [inline checks/raw/binary_artifact_test.go:85]: should you be testing for the content of the results https://github.com/ossf/scorecard/blob/main/checker/raw_result.go#L35?

**Review by azeemshaikh38 — COMMENTED** [review]
Approved by mistake

File: checks/raw/binary_artifact_test.go [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+)
+
+// TestBinaryArtifact tests the BinaryArtifact checker.
+func TestBinaryArtifacts(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		name        string
+		inputFolder string
+		err         error
+		expected    checker.CheckResult
```
Comment (laurentsimon) [inline checks/raw/binary_artifact_test.go:37]: raw part of check does not return`checker.CheckResult` but the results defined in https://github.com/ossf/scorecard/blob/main/checker/raw_result.go#L35. I think we should be testing for this instead, right?

File: checks/raw/binary_artifact_test.go (hunk 1 above)
Comment (naveensrinivasan) [reply checks/raw/binary_artifact_test.go]: Thanks!

File: checks/raw/binary_artifact_test.go (hunk 2 above)
Comment (naveensrinivasan) [reply checks/raw/binary_artifact_test.go]: Yes, agreed took care of it. Now the func checks for the results count.

**Review by azeemshaikh38 — APPROVED** [review]
Awesome, thanks!

**Pull request merged**